src/nullable.ts

```diff
@@ -0,0 +1,35 @@
+import { ModelValueType } from './glossary'
+import { Relation } from './relations/Relation'
+
+export type NullableGetter<ValueType extends ModelValueType> =
+  () => ValueType | null
+
+export class NullableProperty<ValueType extends ModelValueType> {
+  public getValue: NullableGetter<ValueType>
+
+  constructor(getter: NullableGetter<ValueType>) {
+    this.getValue = getter
+  }
+}
+
+export class NullableRelation<RelationType extends Relation<any, any, any>> {
+  public relation: RelationType
+
+  constructor(relation: RelationType) {
+    this.relation = relation
```

Comment:
I think we're extending the base `Relation` class incorrectly for this use case. 

Instead of adding a new `resolveWithNull` method to the base class, override the `resolveWith` method on the `NullableRelation` class. 

> `Relation` on its own doesn't care/know anything about the nullability. It's by the introduction of a child `NullableRelation` class you're creating a subset of the `Relation` that can now handle nullable values. 

```ts
export class NullableRelation extends Relation {
  resolveWith(...args) {
    // Support nullable next values.
  }
}
```

With this change, we can also handle `Relation` and `NullableRelation` instances _the same way_because we don't introduce any custom methods like `resolveWithNull`. Instead, we delegate the nullable support to the child class, keeping the same interface as the parent (which makes sense: when using `NullableRelation` expect `resolveWith` to support nullable next values).

Reply:
**Alternatively**, we can just add the nullable values support straight to the `Relation` class and don't complicate things with the designated `NullableRelation` child. This would largely depend on how complicated the `resolveWihNull` logic turns out to be. If it's as simple as checking if the next value is `null` and allowing that, I'd vote for changing the `Relation` class instead.

Reply:
We resolved this by adding `refs === null` support to the original `Relation` class (its `resolveWith()` method).